### ISSUE TITLE
Hy3 Layout Causes Major Rendering Issues for Chromium/Electron Apps running Native Wayland

### ISSUE THREAD

**Issue by imgurbot12** [outfoxxed/hy3#124]
Reposting after discovering the root cause from: https://github.com/hyprwm/Hyprland/issues/6524

### Description

Brave-Browser and other Ozone Render apps when using Wayland directly as opposed to Xwayland have major issues when using
```
general {
   layout = hy3
}
```

Vesktop, an electron app, fails to launch altogether and dies immediately just hanging indefinitely.

Brave-Browser has a ton of rendering issues when running on native wayland within Hyprland including:
- major pixelation and cropped windowing
- regular tearing and improper rendering
- some tabs failed to render at all and had to be closed and reopened

Here's a short video demonstrating a few of the issues. (I can't seem to upload it properly via github for whatever reason so here's an imgur link)
https://imgur.com/a/O2NcCeK

### How to reproduce

For Vencord:
1. run the following command: `vesktop --ozone-platform=wayland`
2. The gui will begin to launch and then immediately disappear after loading
3. The program will just hang open indefinitely with no ability to interact with it whatsoever

For Brave:

1. Open Brave.
2. Go to `brave://flags`
3. Search for "ozone" and modify "Preffered Ozone Platform" and set it to "wayland"
4. Allow a Restart of the Browser
5. ???
6. Profit
![image](https://github.com/hyprwm/Hyprland/assets/21094168/7f002770-02cb-489b-9b98-5903e8141118)

**Comment by outfoxxed** [comment]
you're sure this is hy3 related? Just tried an electron app under wayland and it seemed to work fine. Also hy3 doesn't intercept window damage.

**Comment by imgurbot12** [comment]
Trying to do some testing and get a clip to confirm but suddenly I'm unable to replicate the same behaviors I posted in the original video.

There were some lingering issues even on dwindle but they seemed greatly reduced immediately after swapping from hy3 to dwindle and I had a harder time replicating on dwindle. Maybe it was just coincidence. I'm very confused now. I have found an issue that exists on both dwindle layout and hy3 layout so I may reopen the issue about that.

https://imgur.com/a/n1Mx0q9
